service/featuregate/README.md

```diff
@@ -0,0 +1,111 @@
+# Collector Feature Gates
+
+This package provides a mechanism that allows operators to enable and disable
+experimental or transitional features at deployment time. These flags should
+be able to govern the behavior of the application starting as early as possible
+and should be available to every component such that decisions may be made
+based on flags at the component level.
+
+## Usage
+
+Feature gates must be defined and registered with the global registry in
+an `init()` function.  This makes the `Gate` available to be configured and 
+queried with a default value of its `Enabled` property.
+
+```go
+var fancyNewFeatureGate = featuregate.Gate{
+	ID:          "namespaced.uniqueIdentifier",
+	Description: "A brief description of what the gate controls",
+	Enabled:     false,
+}
+
+func init() {
+	featuregate.Register(fancyNewFeatureGate)
+}
+```
+
+A `Gate` registered in this manner cannot be queried directly as 
+configuration and CLI settings for feature gates are only applied to a 
+frozen copy of the registry that is made available through the `config.
+Config` associated with a running `service.service` instance.
+
+For components that are started by a `service.service` instance, a read-only 
+implementation of the `featuregate.Gates` interface is available in the 
+`*CreateSettings` structs passed to component factories. It exposes an 
+`IsEnabled(id string) bool` method that can be used to query the status of 
+any `Gate`. 
+
+```go
+if settings.Gates.IsEnabled("namespaced.uniqueIdentifier") {
```

Comment:
Do we need a warning that this does a map lookup and may have (a small) performance impact, so preferably should be queried once? Doesn't matter in vast majority cases, but someone unaware may perform the check in some critical path where the impact may be become visible.